inspirehep/modules/hal/hal_api.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from tempfile import TemporaryFile
+import zipfile
+
+from sword2 import Connection
+from sword2 import http_layer
+import httplib2
+
+from settings import HAL_USER, HAL_PASSWORD
+
+COL_IRI = "https://api-preprod.archives-ouvertes.fr/sword/hal"
+PREPROD = "https://api-preprod.archives-ouvertes.fr/sword/"
+
+
+def add_record(tei, document=None):
+    c = Connection("", user_name=HAL_USER,
+                   user_pass=HAL_PASSWORD,
+                   http_impl=HttpLib2Layer_IgnoreCert(".cache"))
+
+    if document:
+        tempFile = TemporaryFile()
+        with zipfile.ZipFile(tempFile,
+                             mode='w',
+                             compression=zipfile.ZIP_DEFLATED) as zf:
+            zf.writestr('meta.xml', tei)
+            zf.writestr('doc.pdf', document)
+        tempFile.seek(0)
+        payload = tempFile.read()
```

Comment:
Answering myself: yes you can: https://github.com/swordapp/python-client-sword2/blob/master/sword2/connection.py#L727